ingest-app/src/cmr/ingest/services/bulk_update_service.clj

```diff
@@ -56,16 +61,56 @@
        (ingest-events/ingest-collection-bulk-update-event
          task-id concept-id bulk-update-params)))))
 
+(defn- update-collection
+  "Perform the update on the collection"
+  [context concept bulk-update-params]
+  (let [{:keys [update-type update-field find-value update-value]} bulk-update-params
+        update-type (csk/->kebab-case-keyword update-type)
+        update-field (csk/->PascalCaseKeyword update-field)]
+    (field-update/update-concept context concept update-type update-field
+                                 update-value find-value)))
+
+
+(defn- update-and-save-collection
+  "Make collection updates, go through ingest validation. Attempt save to
+  metadata db. If validation or save errors are thrown, they will be caught and
+  handled in handle-collection-bulk-update-event. If warnings were returned,
+  return a warning string."
+  [context concept bulk-update-params]
+  (let [updated-metadata (update-collection context concept bulk-update-params)
+        concept (-> concept
+                    (assoc :metadata updated-metadata)
+                    (update :revision-id inc))
+        ;; If errors are caught, an error will be thrown and logged to the DB
+        ;; If we get warnings back, validation was successful, but will still
+        ;; log warnings
+        {:keys [concept warnings]} (ingest-service/validate-and-prepare-collection
+                                     context concept nil)]
+    (mdb/save-concept context concept)
+    (when (seq warnings)
+      (str "Collection was updated successfully, but translating the collection "
+           "to UMM-C had the following issues: "
+           (str/join "; " warnings)))))
+
 (defn handle-collection-bulk-update-event
   "Perform update for the given concept id. Log an error status of the concept
   cannot be found."
   [context task-id concept-id bulk-update-params]
   (try
     (if-let [concept (mdb2/get-latest-concept context concept-id)]
-      (data-bulk-update/update-bulk-update-task-collection-status context task-id
-        concept-id "COMPLETE" nil)
+      (let [success-status-message (update-and-save-collection context concept bulk-update-params)]
+        (data-bulk-update/update-bulk-update-task-collection-status context task-id
+          concept-id "COMPLETE" success-status-message))
```

Comment:
Caveat: the following response is meant as a conversation starter, not to change your mind in this PR ;-)

I think such an objection to a clear Clojure idiom indicates that there are other problems ... not sure what the underlying issues may be that ultimately make our API calls unclear as to return values, but possibly:

* use better names
* establish conventions for functions with side effects but which also return values
* update docstrings
* include return type metadata in function definitions

It may be that the objection really boils down to the lack of comfort with a language that is not statically typed? Dunno ...

Something else to keep in mind: the use of threading macros are often used for clarity in debugging, as they so clearly show how data passes through the given code.